request_handler.py

```diff
@@ -0,0 +1,177 @@
+# File: request_handler.py
+#
+# Copyright (c) 2016-2022 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+import base64
+import json
+import os
+
+import encryption_helper
+import requests
+from django.http import HttpResponse
+
+from smtp_consts import *
+
+
+def handle_request(request, path_parts):
+    return SMTPRequestHandler(request, path_parts).handle_request()
+
+
+def _get_dir_name_from_app_name(app_name):
+    app_name = ''.join([x for x in app_name if x.isalnum()])
+    app_name = app_name.lower()
+    if not app_name:
+        app_name = "app_for_phantom"
+    return app_name
+
+
+class SMTPRequestHandler:
+    def __init__(self, request, path_parts):
+        self._request = request
+        self._path_parts = path_parts
+        self._rsh = None
+
+    def _return_error(self, error_msg, status):
+        state = self._rsh.load_state()
+        state['error'] = True
+        self._rsh.save_state(state)
+        return HttpResponse(error_msg, status=status, content_type="text/plain")
+
+    def _get_oauth_token(self, code):
+        state = self._rsh.load_state()
+
+        client_id = state['client_id']
+        redirect_uri = state['redirect_url']
+        client_secret = base64.b64decode(state['client_secret']).decode()
+        proxy = state['proxy']
+        token_url = state['token_url']
+
+        if proxy.get('http'):
```

Comment:
Has the proxy setup been fully tested? We have had issues with customers using certain apps that contain proxy code within the app, instead of having the platform handle it. If this has not been tested, we should definitely do so before releasing the app. It should work for both system-wide proxy settings as well as at the app level. Proxy settings are not tested within the pipeline.

Reply:
As the request handler is in different environment than the connector, we had to add the  proxies explicitly. We have tested it with **no-proxy, http-proxy** and **https-proxy**.

Reply:
cool, thanks!